cmd/crank/beta/render/runtime_docker.go

```diff
@@ -222,15 +229,20 @@ func (r *RuntimeDocker) Start(ctx context.Context) (RuntimeContext, error) {
 		r.log.Debug("Container left running", "container", rsp.ID, "image", r.Image)
 		return nil
 	}
-	if r.Stop {
+	if r.Stop || r.Remove {
 		stop = func(ctx context.Context) error {
 			err := c.ContainerStop(ctx, rsp.ID, container.StopOptions{})
 			if err != nil {
 				return errors.Wrap(err, "cannot stop Docker container")
 			}
 
-			err = c.ContainerRemove(ctx, rsp.ID, container.RemoveOptions{Force: true, RemoveVolumes: true})
-			return errors.Wrap(err, "cannot remove Docker container")
+			if r.Remove {
+				err = c.ContainerRemove(ctx, rsp.ID, container.RemoveOptions{Force: true, RemoveVolumes: true})
+				if err != nil {
+					return errors.Wrap(err, "cannot remove Docker container")
+				}
+			}
+			return nil
 		}
```

Comment:
Since we now have multiple cases inside the `stop` function, I'd suggest moving everything in there. Assuming you add the `Cleanup` enum per my other comment, I think something like this would work well:

```go
	stop := func(ctx context.Context) error {
		switch r.Cleanup {
		case AnnotationValueRuntimeDockerCleanupOrphan:
			r.log.Debug("Container left running", "container", rsp.ID, "image", r.Image)
			return nil
		case AnnotationValueRuntimeDockerCleanupStop:
			if err := c.ContainerStop(ctx, rsp.ID, container.StopOptions{}); err != nil {
				return errors.Wrap(err, "cannot stop Docker container")
			}
		case AnnotationValueRuntimeDockerCleanupRemove:
			if err := c.ContainerStop(ctx, rsp.ID, container.StopOptions{}); err != nil {
				return errors.Wrap(err, "cannot stop Docker container")
			}
			if err := c.ContainerRemove(ctx, rsp.ID, container.RemoveOptions{}); err != nil {
				return errors.Wrap(err, "cannot remove Docker container")
			}
		}
		return nil
	}
```

(I know the stop is repeated, but I prefer repeating it to nesting it.)

Reply:
Thanks for the tip, I was thinking something along the lines and I agree with you on the stop repeated but it's better than nesting it